src/lints/trait_associated_const_added.ron

```diff
@@ -0,0 +1,62 @@
+SemverQuery(
+    id: "trait_associated_const_added",
+    human_readable_name: "non-sealed trait added associated constant without default value",
+    description: "A non-sealed trait has gained an associated constant without default value which break downstream implementations of the trait",
+    required_update: Major,
+    lint_level: Deny,
+    reference_link: Some("https://doc.rust-lang.org/cargo/reference/semver.html#trait-new-item-no-default"),
+    query: r#"
+    {
+        CrateDiff {
+            current {
+                item {
+                    ... on Trait {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+
+                        importable_path {
+                            path @output @tag
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+
+                        associated_constant {
+                            associated_constant: name @output @tag
+                            public_api_eligible @filter(op: "=", value: ["$true"])
```

Comment:
This raises an interesting point. I think we don't want this clause, actually.

For `public_api_eligible` to be `false` and get filtered out here, the new const needs to be `#[doc(hidden)]`. But if the trait is not sealed and someone downstream implemented it, adding a new `const` will break their impl _regardless_ of `#[doc(hidden)]` or not. Hiding the `const` will just result in the downstream user being more confused about why their impl is broken. It won't unbreak the impl.

Would you mind adding a test that catches this case and updating the lint?

Reply:
Didn't know public_api_eligible would exclude doc hidden item